.github/workflows/code-ql.yml

```diff
@@ -108,3 +108,27 @@ jobs:
 
     - name: Perform CodeQL Analysis
       uses: github/codeql-action/analyze@v3
+      with:
+        category: "/language:cpp"
+        output: sarif-results
+        upload: failure-only
+
+    - name: filter-sarif
+      uses: advanced-security/filter-sarif@v1
```

Comment:
I think using tags for non-immutable github actions might raise "Unpinned tag for a non-immutable Action in workflow"